dll/win32/shell32/CShellDispatch.cpp

```diff
@@ -119,58 +128,83 @@ HRESULT STDMETHODCALLTYPE CShellDispatch::BrowseForFolder(LONG Hwnd, BSTR Title,
 HRESULT STDMETHODCALLTYPE CShellDispatch::Windows(IDispatch **ppid)
 {
     TRACE("(%p, %p)\n", this, ppid);
+    return CoCreateInstance(CLSID_ShellWindows, NULL, CLSCTX_LOCAL_SERVER, IID_PPV_ARG(IDispatch, ppid));
+}
 
-    *ppid = NULL;
+static HRESULT SHELL_OpenFolder(LPCITEMIDLIST pidl, LPCWSTR Verb = NULL)
+{
+    // Custom function because SHOpenFolderAndSelectItems does not support non-FS pidls nor verbs.
+    SHELLEXECUTEINFOW sei;
+    sei.cbSize = sizeof(sei);
+    sei.fMask = SEE_MASK_IDLIST | SEE_MASK_FLAG_DDEWAIT;
+    sei.hwnd = NULL;
+    sei.lpVerb = Verb;
+    sei.lpFile = sei.lpParameters = sei.lpDirectory = NULL;
```

Comment:
What about zeroing out the whole `sei` variable, then setting just the fields that need to be set?

Reply:
A design choice.

Reply:
For future reference: I'd prefer zeroing out the entire struct, because this also zeroes the padding.
In usermode code this is not as important as kernel code, but is still a good habit to teach yourself.